test/src/materials/NewtonMaterial.C

```diff
@@ -54,12 +53,18 @@ NewtonMaterial::computeQpProperties()
 {
   _p[_qp] = 0.5; // initial guess
 
-  // Newton iteration for find p
-  for (unsigned int i = 0; i < _max_iterations; ++i)
-  {
-    _discrete->computePropertiesAtQp(_qp);
-    _p[_qp] -= _f[_qp] / _f_prime[_qp];
-    if (std::abs(_f[_qp]) < _tol)
-      break;
-  }
+  // Only attempt to solve if iterations are to be taken. This is usually not required, but needed
+  // here to retain the old test behavior that would not trigger a discrete material evaluation. The
+  // NestedSolve class will always evaluate the residual for the initial guess (and will return a
+  // success state if the initial guess was exact).
+  if (getParam<unsigned int>("max_iterations") > 0)
+    _nested_solve.nonlinear(_p[_qp],
+                            // Functor to compute residual and jacobian. The initial guess is not
```

Comment:
_Functor_ is the right term here! Thic does not have to be a _Lambda_, it could be a class with an overloaded `operator()` for example. I'd be happy to change it to `C++ functor (e.g. lambda)` to avoid confusion with MOOSE functors.

Reply:
Oh, it is a lambda here, so nevermind